File: 2025-10/2025-10-03/password_strength.py
def check_strength(password):
    import re
    strength = 0

    if len(password) >= 8:
        strength += 1

    if re.search(r"[a-z]", password) and re.search(r"[A-Z]", password):
        strength += 1

    if re.search(r"\d", password):
        strength += 1

    if re.search(r"[!@#$%^&*]", password):
        strength += 1

    return "weak" if strength < 2 else "medium" if strength < 4 else "strong"

print(check_strength("123456"))
print(check_strength("pass!!!"))
print(check_strength("Qwerty"))
print(check_strength("PASSWORD"))
print(check_strength("PASSWORD!"))
print(check_strength("PassWord%^!"))
print(check_strength("qwerty12345"))
print(check_strength("PASSWORD!"))
print(check_strength("PASSWORD!"))
print(check_strength("S3cur3P@ssw0rd"))
print(check_strength("C0d3&Fun!"))

# --- TEST SUITE ---

tests_text = '''
1. check_strength("123456") should return "weak".
2. check_strength("pass!!!") should return "weak".
3. check_strength("Qwerty") should return "weak".
4. check_strength("PASSWORD") should return "weak".
5. check_strength("PASSWORD!") should return "medium".
6. check_strength("PassWord%^!") should return "medium".
7. check_strength("qwerty12345") should return "medium".
8. check_strength("S3cur3P@ssw0rd") should return "strong".
9. check_strength("C0d3&Fun!") should return "strong".
'''

import re

tests_regex = re.compile(r"(?P<number>\d+)\.\s(?P<function_call>.+) should return (?P<output>.+?)\.?$", re.MULTILINE)
test_data = [item.groupdict() for item in re.finditer(tests_regex, tests_text)]

def run_tests(test_data):
    print("--------------------------")
    print("🧪 Starting Verification...")
    print("--------------------------")

    all_passed = True
    fail_count = 0

    for test in test_data:
        function_call_output = eval(test['function_call'])
        test_output = eval(test['output'])
        
        if function_call_output == test_output:
            print(f"{test['number']}.✅ PASS - Function Call:\n{test['function_call']}")
        else:
            print(f"{test['number']}.❌ FAIL - Function Call:\n{test['function_call']}\nExpected:\n{test_output}\nGot:\n{function_call_output}")
            all_passed = False
            fail_count += 1

    print("----------------------------");

    if (all_passed):
        print("🎉 SUCCESS: All tests PASSED.");
    else:
        print(f"⚠️ WARNING: {fail_count}/{len(test_data)} tests FAILED.");

run_tests(test_data)

File: 2025-10/2025-10-03/passwordStrength.js
function checkStrength(password) {
  let strength = 0;

  if (password.length >= 8)
    strength ++;

  if (/[a-z]/.exec(password) && /[A-Z]/.exec(password))
    strength ++;

  if (/\d/.exec(password))
    strength ++;

  if (/[!@#$%^&*]/.exec(password))
    strength ++;

  return strength < 2 ? "weak" : strength < 4 ? "medium" : "strong";
}

console.log(checkStrength("123456"));
console.log(checkStrength("pass!!!"));
console.log(checkStrength("Qwerty"));
console.log(checkStrength("PASSWORD"));
console.log(checkStrength("PASSWORD!"));
console.log(checkStrength("PassWord%^!"));
console.log(checkStrength("qwerty12345"));
console.log(checkStrength("PASSWORD!"));
console.log(checkStrength("PASSWORD!"));
console.log(checkStrength("S3cur3P@ssw0rd"));
console.log(checkStrength("C0d3&Fun!"));

// --- TEST SUITE ---

const testsText = `
1. checkStrength("123456") should return "weak".
2. checkStrength("pass!!!") should return "weak".
3. checkStrength("Qwerty") should return "weak".
4. checkStrength("PASSWORD") should return "weak".
5. checkStrength("PASSWORD!") should return "medium".
6. checkStrength("PassWord%^!") should return "medium".
7. checkStrength("qwerty12345") should return "medium".
8. checkStrength("S3cur3P@ssw0rd") should return "strong".
9. checkStrength("C0d3&Fun!") should return "strong".
`;

const testsRegex = /(?<number>\d+)\.\s(?<functionCall>.+) should return (?<output>.+?)\.?$/gm;
const testData = [...testsText.matchAll(testsRegex).map(match => match.groups)];

function runTests(testData) {
  console.log("--------------------------");
  console.log("🧪Starting Verification...");
  console.log("--------------------------");

  let allPassed = true;
  let failCount = 0;
  
  testData.forEach(test => {
    const functionCallOutput = eval(test.functionCall);
    const testOutput = eval(test.output);
    let comparison;

    if (Array.isArray(testOutput))
      comparison = arraysEqual(functionCallOutput, testOutput);
    else
      comparison = functionCallOutput === testOutput;

    if (comparison) {
      console.log(`${test.number}.✅PASS - Function Call: ${test.functionCall}`);
    }
    else {
      console.log(`${test.number}.❌FAIL - Function Call: ${test.functionCall}\nExpected: ${testOutput}\nGot: ${functionCallOutput}`);
      allPassed = false;
      failCount ++;
    }
  })

  console.log("----------------------------");

  if (allPassed)
    console.log("🎉SUCCESS: All tests PASSED.");
  else
    console.log(`⚠️WARNING: ${failCount}/${testData.length} tests FAILED.`);
}

function arraysEqual(a, b) {
  if (a.length !== b.length)
    return false;

  for (const index in a) {
    if (Array.isArray(a[index])) {
      if (Array.isArray(b[index])) {
        if (!arraysEqual(a[index], b[index]))
          return false;
      }
      else 
        return false;
    }

    else if (a[index] !== b[index])
      return false;
  }

  return true;
}

runTests(testData);